writer.go

```diff
@@ -99,27 +151,49 @@ func newWriterDictLevel(w io.Writer, cd *CDict, compressionLevel int) *Writer {
 }
 
 // Reset resets zw to write to w using the given dictionary cd and the given
-// compressionLevel.
+// compressionLevel. Use ResetWriterParams if you wish to change other
+// parameters that were set via WriterParams.
 func (zw *Writer) Reset(w io.Writer, cd *CDict, compressionLevel int) {
 	zw.inBuf.size = 0
```

Comment:
Could you init `params` and then call `ResetWriterParams` here? This will remove duplicate code in the `Reset` and `ResetWithParams` functions.

Reply:
done